File: ui/app/types/Environment.ts
export interface Variable {
  key: string;
  value: string;
}

export interface Environment {
  id: string;
  name: string;
  slug: string;
  description?: string;
  variables: Record<string, string>;
  isActive: boolean;
  createdAt: string;
  modifiedAt: string;
}

export type CreateEnvironmentDto = {
  name: string;
  description?: string;
  variables: Record<string, string>;
};

export type UpdateEnvironmentDto = Partial<CreateEnvironmentDto>;

File: api/Innovatek.Parallel.MiniCluster.Core/Entities/Environment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Innovatek.Parallel.MiniCluster.Core.Entities
{
    public class Environment
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
        public required string Slug { get; set; }
        public string? Description { get; set; }

        // Dictionary-based variable storage (from branch)
        public Dictionary<string, string> Variables { get; set; } = new();

        public bool IsActive { get; set; }
    }
}
